src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/workstations/Chef.java

```diff
@@ -0,0 +1,90 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.workstations;
+
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Client;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Order;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.ReadyMeal;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Condiment;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.CookingStrategy;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.DefaultCookingStrategy;
+
+public class Chef implements Observer {
+	
+	private OrderQueue orderQueue;
+	private DeliveryQueue deliveryQueue;
+	private CookingStrategy cookingStrategy = new DefaultCookingStrategy();
+	private boolean isBusy;
+	private int cookingSpeed = 1000;
+
+	public Chef(OrderQueue orderQueue, DeliveryQueue deliveryQueue) {
+		this.orderQueue = orderQueue;
+		this.deliveryQueue = deliveryQueue;
+		orderQueue.addObserver(this);
+	}
+	
+	public Chef(OrderQueue orderQueue, DeliveryQueue deliveryQueue, CookingStrategy cookingStrategy) {
+		this(orderQueue, deliveryQueue);
```

Comment:
Konstruktorláncolás, elegáns. Látom, hogy a szakácsnak van default főzési stílusa, amit a rendelésben megadott stílus felülbírálhat az étel elkészítésének idejére. Felmerül a kérdés, hogy szükség van a lehetőségre, hogy nem default stílusban főző szakácsot hozzunk létre, ha a rendelések úgyis felül fogják ezt bírálni? A kérdés költői, mert nem tudjuk, hogy milyen arányban lesz kitöltve a stílus a rendelésekben, és ha nagyrészt nincs kitöltve, akkor melyik stílus jó nekünk mint default.

Teendő ezzel kapcsolatban nincs. Inkább csak azon gondolkoztam itt el, hogy érdemes-e sok mindent konfigurálhatóvá tenni első körben vagy ráérünk később, amikor felmerül az igény?

Reply:
Igen, ez jó kérdés. Egyelőre úgy tekintettem a főzési stílusra mint egy különleges kérés a client szempontjából, tehát egy default cooking stílus elegendő, amit olykor kivált egy konkrétabb cooking stílus kérés szerint. Lehetne specializált Chef-eket is létrehozni, akik csak egyféleképpen tudnak főzni, és akkor a Order-ek kiosztása is ez alapján történne. Egyelőre most úgy terveztem meg, hogy minden Chef ért minden stílushoz, igazi profik :)

Reply:
:-)